sdk/enter.go

```diff
@@ -153,61 +153,65 @@ func (e *enter) MountAPI() error {
 	return nil
 }
 
-// MountAgent bind mounts a SSH or GnuPG agent socket into the chroot
-func (e *enter) MountAgent(env string) error {
-	origPath := os.Getenv(env)
+// MountAgent bind mounts a SSH agent socket into the chroot
+func (e *enter) mountSSHAgent() error {
+	origPath := os.Getenv("SSH_AUTH_SOCK")
 	if origPath == "" {
 		return nil
 	}
 
-	origDir, origFile := filepath.Split(origPath)
+	origDir := filepath.Dir(origPath)
 	if _, err := os.Stat(origDir); err != nil {
 		// Just skip if the agent has gone missing.
 		return nil
 	}
 
-	newDir, err := ioutil.TempDir(e.UserRunDir, "agent-")
-	if err != nil {
+	newDir := filepath.Join(e.Chroot, origDir)
+	if err := os.Mkdir(newDir, 0700); err != nil && !os.IsExist(err) {
 		return err
 	}
 
-	if err := system.Bind(origDir, newDir); err != nil {
-		return err
-	}
-
-	newPath := filepath.Join(newDir, origFile)
-	chrootPath := strings.TrimPrefix(newPath, e.Chroot)
-	return os.Setenv(env, chrootPath)
+	return system.Bind(origDir, newDir)
 }
 
 // MountGnupg bind mounts $GNUPGHOME or ~/.gnupg and the agent socket
 // if available. The agent is ignored if the home dir isn't available.
-func (e *enter) MountGnupg() error {
+func (e *enter) mountGnupg() error {
 	origHome := os.Getenv("GNUPGHOME")
 	if origHome == "" {
 		origHome = filepath.Join(e.User.HomeDir, ".gnupg")
 	}
 
 	if _, err := os.Stat(origHome); err != nil {
-		// Skip but do not pass along $GNUPGHOME
-		return os.Unsetenv("GNUPGHOME")
```

Comment:
Why are we passing through`GNUPGHOME` in this case?

Reply:
We're not. It's unconditionally unset below.

Reply:
Not after returning it isn't.

Reply:
Oh, hmm. I guess it does make sense to unconditionally unset it before any returns could happen. It's not doing anything with gpg in that case, but it would be set to a path from the real root which might make gpg confused if the user has their own gpg setup inside the sdk.

Looking at this more I'm somewhat confused what is controlling which environment variables are passed through to the chroot and which aren't. If I set `FOOBAR=BARFOO` then `cork enter` and `FOOBAR` is unset. `GNUPGHOME` does stay set though. 😕

Reply:
https://github.com/coreos/mantle/blob/78e6c559eddbcd1653c67078a90d90af72f3e432/sdk/create.go#L67-L75